src/rules/default-import-match-filename.js

```diff
@@ -0,0 +1,120 @@
+import docsUrl from '../docsUrl'
+import isStaticRequire from '../core/staticRequire'
+
+/**
+ * @param {string} filename
+ * @returns {string}
+ */
+function removeExtension(filename) {
+  return filename.replace(/\.[^.]+$/, '')
```

Comment:
```suggestion
  return path.basename(filename, path.extname(filename))
```
with an `import path from 'path'` at the top